src/components/inputs/money-input/money-input.js

```diff
@@ -160,13 +160,13 @@ const createCurrencySelectStyles = ({
 // This means the highest amount always wins. We do this by comparing the last
 // position of `.` and `,`. Whatever occurs later is used as the decimal
 // separator.
-export const getLocalsSperatorAndThrowaway = (rawAmount, locale) => {
+export const getLocalesSperatorAndThrowaway = (rawAmount, locale) => {
   if (locale) {
     const [throwaway, separator] = (9999.999)
-      .toLocaleString(locale)
-      .replace(/9/g, '')
+      .toLocaleString(locale) // after this step it'll be either 9,999.999 or 9.999,999 based in the locale
+      .replace(/9/g, '') // then we remove the number `9` to endup with `,.` or `.,`
       .split('')
-      .map(symbol => (symbol === '.' ? '\\.' : symbol));
+      .map(symbol => (symbol === '.' ? '\\.' : symbol)); // here we escape the '.' to use it as regex
```

Comment:
What is the goal behind the `map`. I still can't follow 100%.

Reply:
it is exactly like https://github.com/commercetools/ui-kit/blob/e65a1ec06baca13aecf5f64673c2bf1b9b7e7f0f/src/components/inputs/money-input/money-input.js#L176

when we remove the throwaway 
https://github.com/commercetools/ui-kit/blob/e65a1ec06baca13aecf5f64673c2bf1b9b7e7f0f/src/components/inputs/money-input/money-input.js#L190

if we didn't do this map, the `.` will be the regex any character, so we escape it